src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -17,6 +17,7 @@ import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
 import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GcsPath, GcsPathSupport, GoogleProject, ServiceAccountKey, ServiceAccountKeyId, parseGcsPath}
 
 final case class ClusterRecord(id: Long,
+                               internalId: String,
```

Comment:
i'm not really happy with this. i would have liked for there to no longer be an `id` field and to just have `internalId` with type `UUID` (although in the DB it would actually be a `VARCHAR(36)` because querying binary fields is horrible).

in order to accomplish that, i'd need to:

1. remove ID field
2. generate UUIDs for every existing cluster in the database and make this the primary key
3. for all existing _active_, generate a new `notebook-cluster` resource in sam. migration for this wouldn't actually be _too_ hairy, i could just use the rawls or agora SAs since they have permission to get access tokens on behalf of users. this works because of the assumption that no user has modified their cluster's policies in sam

since this will fix the problems as-is, i'm not going to go down the rabbit-hole but maybe when i have some free time in the future i will.